flipperdevices/flipperzero-firmware
# [FL-2675] /int space reservation

# What's new

- [FL-2675] Renamed files created in internal storage by services - now they have names starting with "."
- Fixed /int->/ext migration: it mishandled complex tree structure and failed to move nested folders
- Migration now moves all folders & all non-dot files from /int
- Added 5 page space reserved space on /int: storage will refuse to create files & open them for writing when less than 5 pages are left free. These restrictions are not enforced for dot-files
- Added INT_PATH, EXT_PATH & ANY_PATH macros for wrapping FS paths in code
- Added macros for system services (to be used with furi_record_open in place of strings)

# Verification 

- Update from release version with BT keys & desktop settings customization 
- Verify that configuration is preserved after installing this version & that system files have "." in their names
- Place attached .tar archive on SD card
- Unpack that archive with `update restore /ext/T.tar`
- Verify that data migration properly handles nested directories in /int
- Unpack that archive again & try uploading new files to /int, check that upload fails (no free space)
- Try adding new paired device or change desktop settings and verify that these operations don't fail

# Checklist (For Reviewer)


[FL-2675]: https://flipperzero.atlassian.net/browse/FL-2675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
